scheduler/pkg/klcpermit/workflow_manager_test.go

```diff
@@ -73,3 +76,258 @@ func Test_getCRDName(t *testing.T) {
 		})
 	}
 }
+
```

Comment:
please, add also tests for `getCRDName`, `ObserveWorkloadForPod`, and `startWatching`

Reply:
`getCRDName` is already covered by unit tests, but added a missing use-case there, the rest is covered by e2e tests, but will try to add at least a bumch of simple unit tests there, as it may be quite challenging to unit test this